schema/manager.go

```diff
@@ -136,6 +136,68 @@ func (manager *Manager) OrderedSchemas() []*Schema {
 	return res
 }
 
+//reorderSchema tries reorder schemas using Tarjan's algorithm
+type state int
+
+const (
+	notVisited state = iota
+	visited
+	temporaryVisited
+)
+
+func visitSchema(schemas []*Schema, schemaList []string, schema *Schema, visitedSchema map[string]state) ([]string, error) {
+	if visitedSchema[schema.ID] == temporaryVisited {
+		return nil, fmt.Errorf("Schemas aren't DAG. We can't reorder automatically")
+	}
+	if visitedSchema[schema.ID] == visited {
+		return schemaList, nil
+	}
+	visitedSchema[schema.ID] = temporaryVisited
+	relatedSchemas := schema.relatedSchemas()
+	var err error
+	for _, relatedSchemaID := range relatedSchemas {
+		for _, relatedSchema := range schemas {
```

Comment:
Naming `relatedSchema`  here is a bit confusing, because there is "relatedSchamas" just above.
Simply `schema` could be better.